robolectric/src/test/java/org/robolectric/shadows/ShadowBiometricManagerTest.java

```diff
@@ -21,22 +25,78 @@ public class ShadowBiometricManagerTest {
   @Before
   public void setUp() {
     biometricManager =
-            ApplicationProvider.getApplicationContext().getSystemService(BiometricManager.class);
+        ApplicationProvider.getApplicationContext().getSystemService(BiometricManager.class);
     assertThat(biometricManager).isNotNull();
   }
 
   @Test
   public void testCanAuthenticate_serviceNotConnected_canNotAuthenticate() {
-    ((ShadowBiometricManager) Shadow.extract(biometricManager)).setCanAuthenticate(false);
+    ShadowBiometricManager shadowBiometricManager = Shadow.extract(biometricManager);
+    shadowBiometricManager.setCanAuthenticate(false);
 
     assertThat(biometricManager.canAuthenticate())
         .isEqualTo(BiometricManager.BIOMETRIC_ERROR_NO_HARDWARE);
   }
 
   @Test
   public void testCanAuthenticate_serviceConnected_canAuthenticate() {
-    ((ShadowBiometricManager) Shadow.extract(biometricManager)).setCanAuthenticate(true);
+    ShadowBiometricManager shadowBiometricManager = Shadow.extract(biometricManager);
+    shadowBiometricManager.setCanAuthenticate(true);
 
     assertThat(biometricManager.canAuthenticate()).isEqualTo(BiometricManager.BIOMETRIC_SUCCESS);
   }
+
+  @Test
+  public void testCanAuthenticate_serviceNotConnected_noEnrolledBiometric_biometricNotEnrolled() {
+    ShadowBiometricManager shadowBiometricManager = Shadow.extract(biometricManager);
```

Comment:
It's better if we can call `setCanAuthenticate(false)` to ensure service not connected explicitly. And following test methods are expected the same.

Reply:
Sure no problem